tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_cell_data.py

```diff
@@ -17,23 +16,14 @@ def test_enable_cell_data_via_settings_app(self):
 
         settings = Settings(self.marionette)
         settings.launch()
-        cell_and_data_settings = settings.open_cell_and_data()
 
+        cell_and_data_settings = settings.open_cell_and_data()
         self.assertNotEqual(cell_and_data_settings.carrier_name, '')
-
-        # enable cell data
         self.assertFalse(cell_and_data_settings.is_data_toggle_checked)
-        cell_data_prompt = cell_and_data_settings.enable_data()
 
+        cell_data_prompt = cell_and_data_settings.enable_data()
         self.assertTrue(cell_and_data_settings.is_data_toggle_checked)
-        # Even though the data toggle checkbox is already checked,
-        # cell data should not be enabled until we turn it on via the prompt
-        self.assertFalse(self.data_layer.get_setting('ril.data.enabled'),
-            "Cell data was enabled before responding to the prompt")
 
         cell_data_prompt.turn_on()
-
-        # verify that cell data is now enabled and connected
-        self.assertTrue(self.data_layer.is_cell_data_enabled, "Cell data was not enabled via Settings app")
-        Wait(self.marionette).until(lambda m: self.data_layer.is_cell_data_connected,
-            message="Cell data was not connected via Settings app")
+        status_bar = System(self.marionette).status_bar.minimized
+        status_bar.wait_for_data_to_be_connected()
```

Comment:
Yes, we could do that.
You could of the code from here for that, I guess: http://mxr.mozilla.org/gaia/source/tests/python/gaia-ui-tests/gaiatest/tests/functional/cost_control/test_cost_control_data_alert_mobile.py
That test is already testing if self.data_layer.connect_to_cell_data() is working, but like you said, that's an integration test.